src/services/modelRegistry.ts

```diff
@@ -0,0 +1,96 @@
+import { initializeRegistry } from '../context/initializeRegistry';
+import { ChatCompletionTool } from 'openai/resources/index';
+import { ComponentType } from 'react';
+import {
+  CompleteTxDisplay,
+  CompleteTxDisplayProps,
+} from '../features/blockchain/components/displayCards/CompleteTxDisplay';
+import {
+  InProgressQueryDisplay,
+  InProgressQueryProps,
+} from '../features/blockchain/components/displayCards/InProgressQueryDisplay';
+import {
+  CompleteQueryDisplay,
+  InProgressTxDisplay,
+} from '../features/blockchain/components/displayCards';
+import { InProgressTxDisplayProps } from '../features/blockchain/components/displayCards/InProgressTxDisplay';
+import { CompleteQueryDisplayProps } from '../features/blockchain/components/displayCards/CompleteQueryDisplay';
+
+type SupportedBlockchainModels = 'gpt-4o' | 'gpt-4o-mini';
+type SupportedReasoningModels = 'deepseek-chat';
+
+interface ModelConfig {
+  name: SupportedBlockchainModels | SupportedReasoningModels;
+  description: string;
+  tools: ChatCompletionTool[];
+  systemPrompt: string;
+  introText: string;
+  components?: {
+    transaction?: {
+      inProgress: ComponentType<InProgressTxDisplayProps>;
+      complete: ComponentType<CompleteTxDisplayProps>;
+    };
+    query?: {
+      inProgress: ComponentType<InProgressQueryProps>;
+      complete: ComponentType<CompleteQueryDisplayProps>;
+    };
+  };
+}
+
+interface ModelRegistry {
+  blockchain: Record<SupportedBlockchainModels, ModelConfig>;
+  reasoning: Record<SupportedReasoningModels, ModelConfig>;
+}
+
+const registry = initializeRegistry();
```

Comment:
let's rename this or rename `initializeRegistry` to be more specific. Maybe `const messageRegistry = initializeMessageRegistry();`